src/sentry/search/events/fields.py

```diff
@@ -2265,13 +2275,63 @@ def __init__(self, dataset: Dataset, params: ParamsType):
                     ),
                     default_result_type="percentage",
                 ),
+                SnQLFunction(
+                    "percentile",
+                    required_args=[
+                        SnQLColumnOrAlias("column"),
+                        NumberRange("percentile", 0, 1),
+                    ],
+                    snql_aggregate=self._resolve_percentile,
+                    result_type_fn=reflective_result_type(),
```

Comment:
You need a `get_type` [method](https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1219) on your subclass and support [this](https://github.com/getsentry/sentry/blob/master/src/sentry/search/events/fields.py#L1535) I think. So if you do p95 on an integer type field, it will return an integer type and if you do it on a a duration type field (like `transaction.duration` it will result in a duration type.

Reply:
I've added `get_type` to  `SnQLNumericColumnNoLookup`! It's following the original `get_type` from `NumericColumnNoLookup`